src/main.ts

```diff
@@ -7,28 +7,31 @@
 // http://pieroxy.net/blog/pages/lz-string/testing.html
 //
 // LZ-based compression algorithm, version 1.4.5
-var LZString = (function() {
+
+import { LZString, Dictionary, DictionaryCollection, PendingDictionary, DecompressionTracker } from '@/interfaces'
+
+const LZString: LZString = (function() {
 
 // private property
-var f = String.fromCharCode;
-var keyStrBase64 = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
-var keyStrUriSafe = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+-$";
-var baseReverseDic = {};
+const f = String.fromCharCode;
+const keyStrBase64: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+const keyStrUriSafe: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+-$";
+const baseReverseDic: DictionaryCollection = {};
 
-function getBaseValue(alphabet, character) {
+function getBaseValue(alphabet: string, character: string): number {
   if (!baseReverseDic[alphabet]) {
     baseReverseDic[alphabet] = {};
-    for (var i=0 ; i<alphabet.length ; i++) {
+    for (let i=0 ; i<alphabet.length ; i++) {
       baseReverseDic[alphabet][alphabet.charAt(i)] = i;
     }
   }
   return baseReverseDic[alphabet][character];
 }
 
-var LZString = {
-  compressToBase64 : function (input) {
+const LZString: LZString = {
+  compressToBase64 : function (input: string): string {
```

Comment:
The type from the LZString should handle the arguments for these keys - but this also makes it very nicely specific! :-)

Reply:
keeping until post-merge listing pass